creusot/tests/should_succeed/logic_functions.rs

```diff
@@ -17,5 +17,22 @@ logic! {
     fn logical() -> bool { false }
 }
 
+logic! {
+    fn arith(n: Int, b: bool) -> Int {
+        if !b {
+            -n + n - n * n / n
+        } else {
+            n
+        }
+    }
+}
+
+logic! {
+    fn deref_pat<'a>(p: &'a (Int, Int)) -> Int {
+        let (a, b) = p;
```

Comment:
I don't think this is testing `DerefPat`. It should say `(&a,&b)` instead.

Reply:
This is actually using `DerefPat`, turning `& (Int, Int)` into `(&Int, &Int)`.

Reply:
But I'll update this test to pattern-match over `enum`.

Reply:
ahh ok! ignore my comments then.